docs/architecture/adr-030-msg-auth-module.md

```diff
@@ -0,0 +1,222 @@
+# ADR 030: Msg Authorization Module
```

Comment:
The original name was msg delegation but people complained that it was confusing because of validator delegation. So we chose authorization as an alternative

Reply:
so how about the other proposals, Auth Delegation Module?

Reply:
People basically didn't want us to use the word "delegation" at all as I recall.